source/MaterialXCore/Element.h

```diff
@@ -71,6 +71,92 @@ using ElementMap = std::unordered_map<string, ElementPtr>;
 /// A standard function taking an ElementPtr and returning a boolean.
 using ElementPredicate = std::function<bool(ConstElementPtr)>;
 
+/// @class ElemenEquivalenceResult
+/// The results of comparing for equivalence.
+class MX_CORE_API ElementEquivalenceResult
```

Comment:
I'd recommend declaring `ElementEquivalenceResult` and `ElementEquivalenceOptions` _after_ the core `Element` class, since the understanding of these classes depends on understanding Element.  This is more of a consideration for people reading through the code, and hopefully should not impact the compilation of these classes!